pkg/storage/segments/repairer.go

```diff
@@ -157,3 +168,23 @@ func createBucketID(path storj.Path) ([]byte, error) {
 	}
 	return []byte(storj.JoinPaths(comps[0], comps[2])), nil
 }
+
+func (repairer *Repairer) updateMissingPieces(ctx context.Context, pieces []*pb.RemotePiece) (missingPieces []int32, err error) {
```

Comment:
this is identical to the `getMissingPieces` function in `datarepair/checker/chacker.go`
Maybe it would be good to move both to overlay cache and call the same function?